src/main/java/org/opentripplanner/routing/algorithm/strategies/SimpleEuclideanRWH.java

```diff
@@ -0,0 +1,103 @@
+package org.opentripplanner.routing.algorithm.strategies;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.opentripplanner.routing.core.RoutingRequest;
+import org.opentripplanner.routing.core.State;
+import org.opentripplanner.routing.core.vehicle_sharing.*;
+import org.opentripplanner.routing.edgetype.FreeEdge;
+import org.opentripplanner.routing.graph.Edge;
+import org.opentripplanner.routing.graph.Vertex;
+
+import java.util.Optional;
+import java.util.Set;
+
+import static java.lang.Double.min;
+import static org.opentripplanner.common.geometry.SphericalDistanceLibrary.fastDistance;
+
+public class SimpleEuclideanRWH implements RemainingWeightHeuristic {
+
+    private double lat;
+    private double lon;
+    private double bestMultiplier;
+
+    @Override
+    public void initialize(RoutingRequest options, long abortTime) {
+        calculateDestination(options.rctx.target);
+        calculateBestMultiplier(options);
+    }
+
+    private void calculateDestination(Vertex target) {
+        if (target.getDegreeIn() == 1) {
+            Optional<Edge> edge = target.getIncoming().stream().findFirst().filter(FreeEdge.class::isInstance);
+            if (edge.isPresent()) {
+                lat = edge.get().getFromVertex().getLat();
+                lon = edge.get().getFromVertex().getLon();
+                return;
+            }
+        }
+        lat = target.getLat();
+        lon = target.getLon();
+    }
+
+    private void calculateBestMultiplier(RoutingRequest options) {
+        bestMultiplier = options.routingReluctances.getWalkReluctance() / options.walkSpeed;
+        // max speed of transit in not higher than max speed of a car, and transit does not have a reluctance
+        if (options.modes.isTransit()) {
+            bestMultiplier = min(bestMultiplier, 1.0 / options.carSpeed);
+        }
+        if (options.rentingAllowed) {
+            calculateBestMultiplierWithRentableVehicles(options);
+        } else {
```

Comment:
dlaczego else?
nie sprawdzasz starting mode, więc to nie jest tak, że nie mozna wypożyczyć, prawda?
przypomnisz różnicę między `options.carSpeed` i `carDescription.maxPossilbeSpeed()` ?

Reply:
If we have `rentingAllowed = true`, then `startingMode` should be set to `WALK`. If we have renting disabled, then we could have our own vehicle (car or bicycle). We do not mix own vehicles with rentable ones in one request. `options.carSpeed` is max speed of user-owned vehicle and `CarDescription.getMaxPossibleSpeed()` is max speed of rentable car.